opentelemetry-sdk/src/logs/record.rs

```diff
@@ -198,4 +198,53 @@ mod tests {
             Some(vec![(Key::new("key"), AnyValue::String("value".into()))])
         );
     }
+
+    #[test]
+    fn compare_trace_context() {
+        let trace_context = TraceContext {
+            trace_id: TraceId::from_u128(1),
+            span_id: SpanId::from_u64(1),
+            trace_flags: Some(TraceFlags::default()),
+        };
+
+        let trace_context_cloned = trace_context.clone();
+
+        assert_eq!(trace_context, trace_context_cloned);
+
+        let trace_context_different = TraceContext {
+            trace_id: TraceId::from_u128(2),
+            span_id: SpanId::from_u64(2),
+            trace_flags: Some(TraceFlags::default()),
+        };
+
+        assert_ne!(trace_context, trace_context_different);
+    }
+
+    #[test]
+    fn compare_log_record() {
+        let log_record = LogRecord {
+            event_name: Some(Cow::Borrowed("test_event")),
```

Comment:
Maybe also test the equality for `Cow::Borrowed` vs `Cow::Owned`? That would ensure that we treat `Cow::Borrowed("foo")` and `Cow::Owned(String::from("foo"))` as equal and no one mistakenly overrides the `PartialEq` definition to change this behavior in future.

Reply:
Good point! Added.